src/viam/sdk/module/service.cpp

```diff
@@ -271,8 +271,8 @@ void ModuleService::add_api_from_registry_inlock_(API api, const std::lock_guard
     auto new_manager = std::make_shared<ResourceManager>();
 
     const std::shared_ptr<ResourceRegistration> rs = Registry::lookup_resource(api);
-    const std::shared_ptr<ResourceServer> resource_server = rs->create_resource_server(new_manager);
-    resource_server->register_server(server_);
+    const std::shared_ptr<ResourceServer> resource_server =
+        rs->create_resource_server(new_manager, server_);
```

Comment:
I _thought_ I could make the `server_` field of `ModuleService` a `unique_ptr`, but I think we have to share it due to this call (a single `Server` needs to be shared between between potentially multiple `[TypeOfResource]Server` types).

Reply:
I think that depends on whether it is necessary for `[TypeOfResource]Server` objects to potentially extend the lifetime of the `Server` after the `ModuleService` has been destroyed. Does that actually make sense? Otherwise, the `ModuleService` could just own the `Server` object and you could pass it by bare pointer to `[TypeOfResource]Server` constructors. Their lifetime would need to be a subset of the `ModuleService`.

Reply:
> Does that actually make sense? 

I don't think `[TypeOfResource]Server`s should ever outlive the `ModuleService` so that's a good point. I'll try passing by bare pointer.